backend/jobsearch/src/main/kotlin/com/jobsearch/service/VacancyService.kt

```diff
@@ -1,87 +1,111 @@
+    }
 
-        val vacancyEntity = vacancyDto.let {
-            Vacancy(it.id, it.name, it.companyName, it.salaryExpectation, it.yearsOfExperience, it.description, selectedJobFamily)
+    @Transactional
+    fun retrieveAllVacancy(): List<VacancyResponseDTO> {
+        return vacancyRepository.findAll().map {
+            mapToVacancyResponseDto(it)
         }
-        val newVacancy = vacancyRepository.save(vacancyEntity)
-
-        return mapToVacancyDto(newVacancy)
     }
 
-    fun retrieveVacancy(vacancyId: Int): VacancyDto {
-        val vacancy = vacancyRepository.findById(vacancyId)
-            .orElseThrow { NoSuchElementException("No vacancy found with id $vacancyId") }
-
-        return mapToVacancyDto(vacancy)
+    @Transactional
+    fun retrieveVacancyByManager(): List<VacancyResponseDTO> {
+        val manager = userService.retrieveAuthenticatedUser()
+        return vacancyRepository.findByManager(manager).map {
+            mapToVacancyResponseDto(it)
+        }
+    }
+    @Transactional
+    fun findVacanciesByFilter(salary: Int?, jobFamilyId: Int?, yearsOfExperience: Int?): List<VacancyResponseDTO> {
+        val vacancies = vacancyRepository.findVacanciesByFilters(salary, jobFamilyId, yearsOfExperience)
+        return vacancies.map {
+            mapToVacancyResponseDto(it)
+        }
     }
 
-    fun retrieveAllVacancy(): List<VacancyDto> {
-        val persons = vacancyRepository.findAll()
-
-        return persons.map {
-            mapToVacancyDto(it)
+    @Transactional
+    fun createVacancy(vacancyDto: VacancyRequestDTO): VacancyResponseDTO {
+        val selectedJobFamily = jobFamilyService.findByJobFamilyId(vacancyDto.jobFamilyId)
+        val managerUser = userService.retrieveAuthenticatedUser()
+        val vacancyEntity = vacancyDto.let {
+            Vacancy(it.id, it.name, it.companyName, it.salaryExpectation, it.yearsOfExperience, it.description, selectedJobFamily, managerUser)
         }
+
+        val newVacancy = vacancyRepository.save(vacancyEntity)
+
+        return mapToVacancyResponseDto(newVacancy)
     }
 
-    fun updateVacancy(vacancyId: Int, vacancyDto: VacancyDto): VacancyDto {
-        val vacancy = vacancyRepository.findById(vacancyId)
-            .orElseThrow { NoSuchElementException("No vacancy found with id $vacancyId") }
+    @Transactional
+    fun updateVacancy(vacancyId: Int, vacancyDto: VacancyRequestDTO): VacancyResponseDTO {
 
-        val selectedJobFamily = jobFamilyService.findByJobFamilyId(vacancyDto.jobFamilyId!!)
-            .orElseThrow { NoSuchElementException("No vacancy found with id ${vacancyDto.jobFamilyId}") }
+        val vacancy = vacancyRepository.findById(vacancyId)
+            .orElseThrow { NotFoundException("No vacancy found with id $vacancyId") }
+        val manager = userService.retrieveAuthenticatedUser()
+        if (vacancy.manager.email != manager.email ) throw ForbiddenException("You are not allowed to edit this vacancy.")
```

Comment:
is email a unique identifier ok manager?
What will happen if the manager changes the email?
I guess it is better to use `managerId` fr it